src/encoded/static/components/facets/defaults.js

```diff
@@ -101,6 +101,117 @@ DefaultBooleanFacet.contextTypes = {
     navigate: PropTypes.func,
 };
 
+/** Render a binary state boolean facet for the "exists" facet. None of the radio buttons are selected
+ * if the hyperlink does not match YES or NO options.
+ *
+ * YES- <facet>!=*
+ * NO - No facet selected
+ * NEITHER SELECTED - <facet>!=<value> or <facet>=*
+ *
+ * This doesn't get registered as a  default facet component, but is provided so we have one
+ * component to render all "exists" facets
+ * consistently. For future expansion, note that all properties available to facet-rendering
+ * components are available, but this particular implementation only uses a subset of them. */
+export const DefaultExistsBinaryFacet = ({ facet, relevantFilters, queryString, defaultValue }, reactContext) => {
+    let currentOption = defaultValue;
+
+    if (relevantFilters.length === 1) {
+        if (relevantFilters[0].term === '*' && `${facet.field}!` === relevantFilters[0].field) {
+            currentOption = 'yes';
+        } else if (relevantFilters[0].term !== '*') {
+            currentOption = undefined; // neither radio button is selected
+        } else {
+            currentOption = 'no';
+        }
+    } else {
+        currentOption = 'no';
+    }
+
+    const facetYesTerm = facet.terms.find((term) => term.key === 'yes');
+    const facetNOTerm = facet.terms.find((term) => term.key === 'no');
+
+    // This may  be counterintuitive! YES-radio button option means set the facet to "facet!=*", so display
+    // nothing with the facet. The NO-radio button means include all facet values.
+    // So set YES-radio button to the facet-no count and set the NO-radio button to both the combined
+    // facet-yes and facet-no counts
+    const terms = [
+        { key: 'yes', doc_count: facetNOTerm.doc_count },
+        { key: 'no', doc_count: facetYesTerm.doc_count + facetNOTerm.doc_count },
+    ];
+
+    // We have to build the new query string
+    // which uses the `remove` link from the relevant filter. This callback gets memoized to avoid
+    // needlessly re-rendering this component, and its dependencies should normally not change until
+    // the user clicks a term.
+    const handleRadioClick = React.useCallback((event) => {
```

Comment:
It’d be interesting to compare this with not memoizing. I don’t think memoizing this helps with preventing rerenders because you’re passing `handleRadioClick` to an `<input>` DOM component which gets rendered regarldess of whether its properties change or not. Memoizing can help when passing the memoized thing to a custom component, and then only in some cases.